packages/playground/src/components/tf_notification.vue

```diff
@@ -0,0 +1,46 @@
+<template>
+  <div></div>
+</template>
+
+<script lang="ts" setup>
+import "mosha-vue-toastify/dist/style.css";
+
+import { ContractStates } from "@threefold/grid_client";
+import { createToast } from "mosha-vue-toastify";
+import { onMounted } from "vue";
+import { ref } from "vue";
+
+import { useProfileManager } from "../stores";
+import { getUserContracts, type NormalizedContract } from "../utils/contracts";
+import { getGrid } from "../utils/grid";
+
+const profileManager = useProfileManager();
+const contracts = ref<NormalizedContract[]>([]);
+
+onMounted(async () => {
+  contracts.value = [];
+
+  while (profileManager.profile) {
+    const grid = await getGrid(profileManager.profile!);
+    contracts.value = await getUserContracts(grid!);
```

Comment:
I think it's easier to get the contract in the grace period instead of getting them, then start filtering them.
you can do that by 
`gridClient.contracts.listMyContracts({state: [ContractStates.GracePeriod]})`